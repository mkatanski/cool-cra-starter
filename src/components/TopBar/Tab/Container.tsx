import styled from "styled-components";

export type TabProps = {
  alignRight?: boolean;
};

export const Container = styled.div<TabProps>`
  height: 100%;
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  justify-content: center;
  align-content: space-between;
  align-items: flex-start;

  margin-left: ${(props) => (props.alignRight ? "auto" : "inherit")};
`;
