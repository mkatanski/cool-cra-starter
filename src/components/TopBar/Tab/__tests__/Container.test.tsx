import React from "react";
import TestRenderer from "react-test-renderer";

/* Uncomment below if needed */
// import { render, getByTestId } from "@testing-library/react";

import { Container } from "../Container";

/* Uncomment below if You need/want to use enzyme */
// import enzyme from "enzyme";

/* Component mock example */
// jest.mock("[module]", () => ({
//   [ComponentName]: ({children, ...rest}: React.PropsWithChildren<{}>) => (
//     <div id="[ComponentName]" {...rest}>{children}</div>
//   )
// }));

describe("components", () => {
  describe("TopBar.Container", () => {
    it("should render Container component", () => {
      const tree = TestRenderer.create(<Container />);

      expect(tree).toMatchSnapshot();
    });
  });
});
