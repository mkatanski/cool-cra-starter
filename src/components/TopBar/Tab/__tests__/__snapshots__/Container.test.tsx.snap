// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components TopBar.Container should render Container component 1`] = `
.c0 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-content: space-between;
  -ms-flex-line-pack: space-between;
  align-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  margin-left: inherit;
}

<div
  className="c0"
/>
`;
